ext/standard/array.c

```diff
@@ -6419,7 +6419,7 @@ PHP_FUNCTION(array_combine)
 	num_values = zend_hash_num_elements(values);
 
 	if (num_keys != num_values) {
-		php_error_docref(NULL, E_WARNING, "Both parameters should have an equal number of elements");
+		php_exception_or_error_docref(NULL, E_WARNING, "Both parameters should have an equal number of elements");
```

Comment:
I believe it's okay to convert this kind of error condition to an Error unconditionally (I've been doing this occasionally). Basically any error condition that indicates a programming error and no reasonable code should handle locally.

For this kind of directive, I think the area to focus on would be things like I/O errors.

Reply:
Okay, what is the best way to convert these? Using ``zend_throw_exception(zend_ce_error_exception, message, E_ERROR);`` directly instead or is there a predefined function/macro for throwing ErrorExceptions?

Reply:
I personally think these are not strictly _programmer errors_, and could save us from an awful lot of boilerplate pre-validation code if e.g. the arrays are populated from some file or even a HTTP request.